plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -62,14 +68,27 @@
 
   // Preference store (should be configuration scoped) from which we get UUID, opt-in status, etc.
   private IEclipsePreferences preferences;
-
-  private OptInDialogCreator optInDialogCreator;
+  private Display display;
+
+  private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
Is it necessary to declare it `ConcurrentLinkedQueue`? Cannot it be simply `Queue`? Of course it will be instantiated as `ConcurrentLinkedQueue`. If `Queue` is used, then the `@VisibleForTesting` constructor can also use Queue as parameter.